### PR TITLE
crypto: Add `OlmMachine::query_keys_for_users`

### PR DESCRIPTION
**Pull request by richvdh**
Sometimes we need our key query results to be as up-to-date as possible. Add a mechanism to allow that.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 1]
```diff
@@ -402,6 +402,30 @@ impl OlmMachine {
...
+    ///
+    /// * `users` - list of users whose keys should be queried
+    ///
+    /// # Returns
+    ///
+    /// A request to be sent out to the server. Once sent, the response should
+    /// be passed back to the state machine using [`mark_request_as_sent`].
+    ///
+    /// [`mark_request_as_sent`]: #method.mark_request_as_sent
+    /// [`get_identity`]: #method.get_identity
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/machine.rs:420]:
```suggestion
    /// [`get_identity`]: OlmMachine::get_identity
```

**Review by poljar — APPROVED** [review]
Left some suggestions, otherwise looks good.

**Comment by richvdh** [comment]
This closes https://github.com/matrix-org/matrix-rust-sdk/issues/2263.

**Comment by jplatte** [comment]
> This closes https://github.com/matrix-org/matrix-rust-sdk/issues/2263.

Add that to the PR description, and merging will automatically close the issue. Doesn't work from comments.

**Pull request merged**